backend/app/src/main/java/com/moh/phlat/backend/repository/ProcessDataRepository.java

```diff
@@ -33,4 +35,413 @@ public interface ProcessDataRepository extends JpaRepository<ProcessData, Long>
+	@Query(value="Select distinct MAIL_CITY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailCity(Long controlTableId);
+	@Query(value="Select distinct MAIL_BC FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailBc(Long controlTableId);
+	@Query(value="Select distinct MAIL_PCODE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailPcode(Long controlTableId);
+	@Query(value="Select distinct MAIL_COUNTRY FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailCountry(Long controlTableId);
+	@Query(value="Select distinct MAIL_ADDR_IS_PRIVATE FROM process_data WHERE control_id = :controlTableId",nativeQuery=true)
+	public List<String> findAllDistinctMailAddrIsPrivate(Long controlTableId);
+	
+	public static Specification<ProcessData> buildSpecificationIn(Long controlId, String reqRowStatusCode, List<String> ids, List<String> actions, 
+			List<String> rowStatusCode, List<String> messages, List<String> doNotLoad, List<String> stakeholder, List<String> hdsLpcId, 
+			List<String> hdsCpnId, List<String> hdsProviderId1, List<String> hdsProviderId2, List<String> hdsProviderId3, 
+			List<String> hdsProviderIdType1, List<String> hdsProviderIdType2, List<String> hdsProviderIdType3, List<String> hdsHibcFacId, 
+			List<String> hdsType, List<String> hdsName, List<String> hdsNameAlias, List<String> hdsPrefNameFlag, List<String> hdsEmail, 
+			List<String> hdsWebsite, List<String> hdsBusTelAreaCode, List<String> hdsBusTelNum, List<String> hdsTelExt, List<String> hdsCellAreaCode,
+			List<String> hdsCellNum, List<String> hdsFaxAreaCode, List<String> hdsFaxNum, List<String> hdsServiceDelType, List<String> pcnCLinicType,
+			List<String> pcnPciFlag, List<String> hdsHoursOfOp, List<String> hdsContactName, List<String> hdsIsForProfitFlag, List<String> sourceStatus,
+			List<String> hdsParentIpcId, List<String> busIpcId, List<String> busCpnId, List<String> busName, List<String> busLegalName, 
+			List<String> busPayeeNum, List<String> busOwnerName, List<String> busOwnerType, List<String> busOwnerTypeOther, 
+			List<String> facBuildingName, List<String> facHdsDetailAddInfo, List<String> physAddr1, List<String> physAddr2, List<String> physAddr3,
+			List<String> physAddr4, List<String> physCity, List<String> physProv, List<String> physPCode, List<String> physCountry, 
+			List<String> physAddrIsPrivate, List<String> mailAddr1, List<String> mailAddr2, List<String> mailAddr3, List<String> mailAddr4, 
+			List<String> mailCity, List<String> mailBc, List<String> mailPcode, List<String> mailCountry, List<String> mailAddrIsPriv) {
+		
+		Specification<ProcessData> combinedSpecification = Specification.where((root, query, builder) -> 
+				builder.equal(root.get("controlTableId"), controlId));
+
+		if(reqRowStatusCode != null) {
```

Comment:
a better readable code could be to write a command method to check null and add specification as "and"
example - addSpecificationIfNotNull(specs, "createdBy", createdBy); 

and common method could be 
```
private static void addSpecificationIfNotNull(List<Specification<Control>> specs, String fieldName, List<String> values) {
    if (values != null) {
        specs.add((root, query, builder) -> root.get(fieldName).in(values));
    }
}
```